### DISCUSSION TITLE
The order of files in the editor does not reflect logical order

### DISCUSSION THREAD

**Discussion by sacrosanctic** [sveltejs/kit#6046] [General]
The logical order of files should be:
```
.server.js
.js
.svelte
```
but that is not lined up alphabetically. Are there any plans to adjust for this?

This logic follows the flow of data reflected in the doc and Rich's comments.
> In other words +page.server.js passes data along to +page.js, which passes data along to +page.svelte.
https://kit.svelte.dev/docs/load#input-properties-data

>
> ```diff
> src/routes/
> ├ my-api-route/
> + └ +server.js
> ├ my-page-route/
> │ ├ +error.svelte
> + ├ +layout.server.js
> │ ├ +layout.js
> │ ├ +layout.svelte
> + ├ +page.server.js
> │ ├ +page.js
> │ └ +page.svelte
> ```
>https://github.com/sveltejs/kit/discussions/5748#discussioncomment-3282732

**Comment by rmunn** [comment]
Non-serious suggestion: use Typescript, because `.ts` comes after `.server` alphabetically! :grin:

But seriously, 1) that's a terrible reason to make a major language switch, and 2) it still wouldn't solve the specific issue you're asking for, because `.ts` also comes after `.svelte` alphabetically so the order would end up being wrong anyway.

**Reply by sacrosanctic** [reply]
Thanks for the reply, I just wanted to see another person's take on it. I hadn't thought about using `.ts`. 

It's not that serious of a take since I don't have a proposal for a fix.
